### ISSUE TITLE
Response has file-> url but still error uploading message 

### ISSUE THREAD

**Issue by harshalone** [editor-js/image#61]
Hi @neSpecc @gohabereg  

The response has file->url but still error uploading message 

What I am doing wrong can anyone help?
![Image of Error](https://i.vgy.me/U98iFP.png)

**Comment by harshalone** [comment]
Live preview
Go to: https://www.blognow.org/home

Click on Getin and then Login

Click on menu and then Articles

Click on write story

**Comment by harshalone** [comment]
this is my editors configuration:
```

const editor = new EditorJS({
  /**
   * Id of Element that should contain Editor instance
   */
  holder: 'heditor', 

  /** 
   * Available Tools list. 
   * Pass Tool's class or Settings object for each Tool you want to use 
   */ 
	  tools: { 
		header: {
		  class: Header, 
		  inlineToolbar: ['link'],
		  config: {
            placeholder: 'Title...'
          }
		}, 
		paragraph: {
		  class: Paragraph,
		  inlineToolbar: true,
		},
		image: {
		  class: ImageTool,
		  config: {
			endpoints: {
			  accept: 'image/*',
			  byFile: $('#heditor').data('url'), // Your backend file uploader endpoint
			  byUrl: $('#heditor').data('url'), // Your endpoint that provides uploading by Url
			},
			additionalRequestHeaders : {
			  'x-auth-token': $('#heditor').data('token'), 
			} 
		  }
		},
		linkTool: {
		  class: LinkTool,
		  config: {
			endpoint: 'http://localhost:8008/fetchUrl', // Your backend endpoint for url data fetching
		  }
		},
		inlineCode: {
		  class: InlineCode,
		  shortcut: 'CMD+SHIFT+M',
		},
	    Marker: {
		  class: Marker,
		  shortcut: 'CMD+SHIFT+M',
		},
		embed: Embed,
		
		list: { 
		  class: List, 
		  inlineToolbar: true 
		} 
	  }, 
	  
	  /**
       * This Tool will be used as default
       */
      // initialBlock: 'paragraph',

      /**
       * Initial Editor data
       */
      data: {
        blocks: [
          {
            type: "header",
            data: {
              text: "Title of your story",
              level: 2
            }
          },
          {
            type : 'paragraph',
            data : {
              text : 'Hey. Meet the new Story Editor. On this page you can start writing your story â€” try to edit this text...'
            }
          }  
        ]
      },
      onReady: function(){
        saveButton.click();
      },
      onChange: function() {
        console.log('something changed');
      }
});

/**
     * Saving example
     */
    saveButton.addEventListener('click', function () {
      editor.save().then((savedData) => {
         $("#output").html(savedData);
      });
    });
```

**Comment by neSpecc** [comment]
@harshalone what do you see at the console? There should be an error message

**Comment by y3d3n** [comment]
@harshalone how did you manage backend for uploading img

**Comment by wutsi** [comment]
@harshalone Im having the same issue.
Did you manage to resolve it? if yes, how did you do it?
Thanks

**Comment by y3d3n** [comment]
@wutsi the problem is solved. it was so easy.
I'm using laravel as a backend this might help to someone.
this is a controller to store photo

`public function store(Request $request)
    {

        $image = $request->file('image');

        $new_name = rand(). '.' .$image->getClientOriginalExtension();

        $image->move(public_path("img/upload/"), $new_name);

        
        return response()->json(['success'=>'1', 'file'=>['url'=>'/img/upload/'.$new_name]]);
    }`
I'm using 'image' field name as it is the default name of uploaded image field in POST request.

config: {
                endpoints: {
                    accept: 'image/*',
                    byFile: 'http://127.0.0.1:8000/photos/',
                },
                additionalRequestHeaders :{
                    'X-Requested-With': 'XMLHttpRequest',
                    'X-CSRF-TOKEN' : token,
                }
                
            }
and this is the config for the image tool.

**Comment by wutsi** [comment]
Thank you! I'll try this @y3d3n

**Comment by antonyleme** [comment]
> @wutsi the problem is solved. it was so easy.
> I'm using laravel as a backend this might help to someone.
> this is a controller to store photo
> 
> `public function store(Request $request)
> {
> 
> ```
>     $image = $request->file('image');
> 
>     $new_name = rand(). '.' .$image->getClientOriginalExtension();
> 
>     $image->move(public_path("img/upload/"), $new_name);
> 
>     
>     return response()->json(['success'=>'1', 'file'=>['url'=>'/img/upload/'.$new_name]]);
> }`
> ```
> 
> I'm using 'image' field name as it is the default name of uploaded image field in POST request.
> 
> config: {
> endpoints: {
> accept: 'image/*',
> byFile: 'http://127.0.0.1:8000/photos/',
> },
> additionalRequestHeaders :{
> 'X-Requested-With': 'XMLHttpRequest',
> 'X-CSRF-TOKEN' : token,
> }
> 
> ```
>         }
> ```
> 
> and this is the config for the image tool.

I'm having cors issue. What could it be?

My cors middleware in laravel wich just doesnt work for this case:

```
public function handle($request, Closure $next) {

        return $next($request)
          ->header('Access-Control-Allow-Origin', '*')
          ->header('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS')
          ->header('Access-Control-Allow-Headers',' Origin, Content-Type, Accept, Authorization, X-Request-With')
          ->header('Access-Control-Allow-Credentials',' true');
    }
```

And my image config:
```
config: {
     endpoints: {
             accept: 'image/*',
             byFile: 'http://127.0.0.1:8000/api/pop/image/upload',
     },
     additionalRequestHeaders :{
            'X-Requested-With': 'XMLHttpRequest',
     }
}
```

Cors error:

> Access to XMLHttpRequest at 'http://127.0.0.1:8000/api/pop/image/upload' from origin 'http://localhost:3000' has been blocked by CORS policy: Request header field x-requested-with is not allowed by Access-Control-Allow-Headers in preflight response.

**Comment by dottodot** [comment]
I'm having the same issue, using ExpressJS my file is uploading and the response looks to be correct, I do get an message in console but it's not really clear as to what the issue is.

![Screenshot 2020-06-12 at 10 34 17](https://user-images.githubusercontent.com/811652/84488694-79ff4380-ac98-11ea-9e17-d1539e34ab89.png)
![Screenshot 2020-06-12 at 10 35 02](https://user-images.githubusercontent.com/811652/84488706-7d92ca80-ac98-11ea-8f3a-7205ad294c85.png)

**Comment by y3d3n** [comment]
> I'm having the same issue, using ExpressJS my file is uploading and the response looks to be correct, I do get an message in console but it's not really clear as to what the issue is.
> 
> ![Screenshot 2020-06-12 at 10 34 17](https://user-images.githubusercontent.com/811652/84488694-79ff4380-ac98-11ea-9e17-d1539e34ab89.png)
> ![Screenshot 2020-06-12 at 10 35 02](https://user-images.githubusercontent.com/811652/84488706-7d92ca80-ac98-11ea-8f3a-7205ad294c85.png)

is your image uploading?

**Comment by dottodot** [comment]
Yes the image uploads and I checked the image was accessible, I found that using `uploadByFile` worked instead.
